runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/layouts/metrics/HistogramsLayout.java

```diff
@@ -125,18 +126,20 @@ protected int recordSize()
     }
 
     // exclusive upper limits of each bucket
-    private static Map<Integer, Long> generateBucketLimits()
+    private static long[] generateBucketLimits()
     {
-        Map<Integer, Long> limits = new HashMap<>();
+        List<Long> limits = new ArrayList<>();
         for (int i = 0; i < BUCKETS; i++)
         {
-            limits.put(i, 1L << (i + 1));
+            limits.add(1L << (i + 1));
         }
-        return limits;
+        return limits.stream().mapToLong(l -> l).toArray();
```

Comment:
This can be simplified to `new long[BUCKETS]` instead of using `List<Long>`, `.stream().map(...).toArray()`.

Reply:
sure, done.